dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/utils/ByteArrayExt.kt

```diff
@@ -42,7 +42,8 @@ internal fun Collection<ByteArray>.join(
 ): ByteArray {
     val result = ByteArray(
         this.sumOf { it.size } +
-            prefix.size + suffix.size + separator.size * (this.size - 1)
+            prefix.size + suffix.size +
+            if (this.isNotEmpty()) separator.size * (this.size - 1) else 0
```

Comment:
This begins to be hard to read, it'd be nice to split that into variables, e.g.: 

```
val dataSize = this.sumOf { it.size }
val separatorsSize = if (this.isNotEmpty()) separator.size * (this.size - 1) else 0
val size = dataSize + separatorsSize + prefix.size + suffix.size
val result = ByteArray(size)
```